src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/service/PeriodLengthDomainEventService.kt

```diff
@@ -0,0 +1,99 @@
+package uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.service
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.domain.event.EventSource
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.domain.event.HmppsPeriodLengthMessage
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.domain.event.PersonReference
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.domain.event.PersonReferenceType
+import java.time.ZonedDateTime
+
+@Service
+class PeriodLengthDomainEventService(
+  private val snsService: SnsService,
+  @Value("\${ingress.url}") private val ingressUrl: String,
+  // TODO this lookup path doesnt exist yet?? will be covered by another ticket I think then this TODO can be resolved
```

Comment:
Just highlighting this comment in the code - is this the plan? i.e. create period-length endpoints in a separate ticket?